packages/telegram-stamper/src/index.ts

```diff
@@ -0,0 +1,185 @@
+            );
+          }
+        }
+      );
+    } else {
+      // attempt to get creds from telegram cloud storage
+      await window.Telegram.WebApp.CloudStorage.getItem(
+        this.getCloudStorageKeyName(),
+        (err: any, apiKey: string) => {
+          if (err != null || !apiKey) {
+            throw new TelegramStamperError(
+              `Failed getting api key from Telegram Cloud Storage${
+                err && `: ${err}`
+              }`
+            );
+          }
+
+          const { parsedPublicKey, parsedPrivateKey } =
+            this.parseAPIKey(apiKey);
+
+          // ToDo: api key validation, how does apikeystamper validate creds passed to it?'
+
+          if (!parsedPublicKey || !parsedPrivateKey) {
+            throw new TelegramStamperError(
+              "Failed parsing api key from Telegram Cloud Storage"
+            );
+          }
+
+          // instantiate api key stamper
+          this.stamper = new ApiKeyStamper({
+            apiPublicKey: parsedPublicKey,
+            apiPrivateKey: parsedPrivateKey,
+          });
+        }
+      );
+    }
+  }
+
+  async stamp(payload: string) {
+    // check to see if were in a telegram mini app context
+    this.checkTelegramContext();
+
+    // check to see that the stamper was initialized
+    if (!this.stamper) {
+      throw new TelegramStamperError(
+        "Cannot stamp with unintialized telegram stamper, call TelegramStamper.init()"
+      );
+    }
+
+    return await this.stamper.stamp(payload);
+  }
+
+  // clear key from telegram cloud storage
+  async clearKey(key: string) {
+    // check to see if were in a telegram mini app context
+    this.checkTelegramContext();
+
+    await window.Telegram.WebApp.CloudStorage.removeItem(
+      key,
+      (err: any, removed: boolean) => {
+        if (err || !removed) {
+          throw new TelegramStamperError(
+            `Failed removing key: ${key}${err && `: ${err}`}`
+          );
+        }
+      }
+    );
+  }
+
+  checkTelegramContext() {
+    if (window?.Telegram?.WebApp == null) {
+      throw new TelegramStamperError(
+        "Cannot use telegram stamper in non telegram mini-app environment"
+      );
+    }
+  }
+
+  getCloudStorageKeyName() {
+    return TURNKEY_CLOUD_STORAGE_KEY + this.cloudStorageKeySuffix;
+  }
+
+  stringifyAPIKey() {
+    return JSON.stringify({
+      apiPublicKey: this.stamper?.apiPublicKey,
+      apiPrivateKey: this.stamper?.apiPrivateKey,
+    });
+  }
+
+  parseAPIKey(apiKey: string) {
+    try {
+      const parsedApiKey = JSON.parse(apiKey);
+
+      if (!this.isApiKey(parsedApiKey)) {
+        return {
+          parsedPublicKey: "",
+          parsedPrivateKey: "",
+        };
+      }
+
+      return {
+        parsedPublicKey: parsedApiKey.apiPublicKey,
+        parsedPrivateKey: parsedApiKey.apiPrivateKey,
+      };
+    } catch (err) {
+      throw new TelegramStamperError(
+        "Failed parsing api key from Telegram Cloud Storage"
```

Comment:
nit: API (capitalize)

Reply:
fixed a bunch of these in error messages